news/views.py

```diff
@@ -326,6 +326,12 @@ def event(self):
             return get_object_or_404(Event, pk=self.kwargs["event_pk"])
         return None
 
+    def get(self, request, *args, **kwargs):
```

Comment:
Use [LoginRequiredMixin](https://docs.djangoproject.com/en/2.2/topics/auth/default/#the-loginrequired-mixin) instead. Then you don't have to check this and the same thing further down.

Reply:
That is better.